### ISSUE TITLE
[bug] npm 5.5.1 Cannot find module 'internal/util/types'

### ISSUE THREAD

**Issue by dshemin** [npm/npm#19063]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

All npm command produce same error:
```
npm ERR! code MODULE_NOT_FOUND
npm ERR! Cannot find module 'internal/util/types'
```

Log file:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'cache', 'verify' ]
2 info using npm@5.5.1
3 info using node@v9.0.0
4 verbose config Skipping project config: /home/freedemster/.npmrc. (matches userconfig)
5 verbose npm-session 811164f068063b15
6 verbose stack Error: Cannot find module 'internal/util/types'
6 verbose stack     at Function.Module._resolveFilename (module.js:542:15)
6 verbose stack     at Function.Module._load (module.js:472:25)
6 verbose stack     at Module.require (module.js:585:17)
6 verbose stack     at require (internal/module.js:11:18)
6 verbose stack     at evalmachine.<anonymous>:31:26
6 verbose stack     at Object.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/cmd-shim/node_modules/graceful-fs/fs.js:11:1)
6 verbose stack     at Module._compile (module.js:641:30)
6 verbose stack     at Object.Module._extensions..js (module.js:652:10)
6 verbose stack     at Module.load (module.js:560:32)
6 verbose stack     at tryModuleLoad (module.js:503:12)
6 verbose stack     at Function.Module._load (module.js:495:3)
6 verbose stack     at Module.require (module.js:585:17)
6 verbose stack     at require (internal/module.js:11:18)
6 verbose stack     at Object.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/cmd-shim/node_modules/graceful-fs/graceful-fs.js:3:27)
6 verbose stack     at Module._compile (module.js:641:30)
6 verbose stack     at Object.Module._extensions..js (module.js:652:10)
7 verbose cwd /home/freedemster/projects/php/async/react/node/post
8 verbose Linux 4.13.7-1-ARCH
9 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "cache" "verify"
10 verbose node v9.0.0
11 verbose npm  v5.5.1
12 error code MODULE_NOT_FOUND
13 error Cannot find module 'internal/util/types'
14 verbose exit [ 1, true ]
```

#### How can the CLI team reproduce the problem?

Run any commands such as `npm install`, `npm cache` or etc.

### supporting information:

 - `npm -v` prints: 5.5.1
 - `node -v` prints: 9.0.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Linux Arch 4.13.7-1
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by antrez770** [comment]
Hi I wrote about same problem 4 days ago in Windows 10 - https://github.com/npm/npm/issues/19032

**Comment by Basputaek13** [comment]
Download config thank you

**Comment by antrez770** [comment]
Hi What config?

**Comment by antrez770** [comment]
If some body will find the way please write the solution to https://github.com/npm/npm/issues/19032

**Comment by doveraudio** [comment]
I deleted the npm root folder, uninstalled node, and then reinstalled from main node site. my issue is resolved.

**Comment by dshemin** [comment]
> If some body will find the way please write the solution to #19032

If you find a solution, please write in this thread too.

> I deleted the npm root folder, uninstalled node, and then reinstalled from main node site. my issue is resolved.

I think it's not a proper fix for this bug.

**Comment by stavmil** [comment]
For macs
Tried changing to different node versions as some posts suggested in other threads, didn't work.

Ran `brew upgrade node` and it fixed the problem.

**Comment by Chaucharian** [comment]
Is there some solution ? I'm not alone at least :)

**Comment by drball** [comment]
I upgraded node to 9.0.0 and npm to 5.5.1 and I get this error too.

**Comment by DaSchTour** [comment]
I've the same problem with node 8.8.1 and npm 5.4.2 😕

**Comment by wanqing19954** [comment]
So do I, My node -v8.9.1 and npm 5.5.1

**Comment by antrez770** [comment]
Wanqing19954 what OS do You use?

**Comment by wanqing19954** [comment]
macOS 10.12.6

**Comment by WaleedAshraf** [comment]
getting same issue on node@8.9.1 and npm@5.5.1

**Comment by tomchiverton** [comment]
Same here. macOS 10.12.6, latest nodejs LTS.

**Comment by rystraum** [comment]
I've been able to run my app again by downgrading to npm 5.5.0

**Comment by boltex** [comment]
same issue

**Comment by erdemalpay** [comment]
I'm on ubuntu and having the same issue on node@8.9.1 and npm@5.5.1. Downgrading npm to 5.5.0 didn't worked.
I tried installing but neither package manager install nor nvm install didn't worked. I'm still having the issue on ubuntu.

**Comment by mmilenkovic** [comment]
After upgrade to OS X 10.13.1 it started failing.

brew upgrade node - fixed the issue

**Comment by WaleedAshraf** [comment]
Fixed!
If any of your dependency or peer dependency is using **graceful-fs <  4.x** , you need to upgrade it to latest version graceful-fs@4.1.11

**Comment by jamessandoval** [comment]
WaleedAshraf Thank you. This worked for me on mac, express-handlebars ended up having a graceful-fs@3x. Updated to > 4x and all is working. I ran a grep -nr 'graceful-fs' node_modules to find the culprit and everything is running smoothly.

**Comment by johnsonsamuel** [comment]
Same error :(

node - v8.9.1

npm - 5.5.1

error 👍 
Error: Cannot find module 'internal/util/types'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at evalmachine.<anonymous>:31:26

Any help appreciated :)

**Comment by kenany** [comment]
Duplicate of https://github.com/npm/npm/issues/19032.

**Comment by phpcodexpert** [comment]
How we fix it node js v8.9.1?
Error: Cannot find module 'C:\xampp\htdocs\node_modules\npm\phantomjs'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Function.Module.runMain (module.js:676:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3

**Comment by Asalan** [comment]
@WaleedAshraf how to upgrade graceful-fs ??
Are you using Mac OS V 10.13.1 ?

**Comment by WaleedAshraf** [comment]
@Asalan update the package version.

**Comment by PriyankaBondarde** [comment]
how To fix this issue?

Error: Cannot find module 'C:\Users\Admin\Desktop\App_Base_Structure\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Function.Module.runMain (module.js:676:10)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
module.js:538
    throw err;
    ^

**Comment by farrukh-malik** [comment]
Error: Cannot find module 'internal/util/types'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at evalmachine.<anonymous>:31:26
    at Object.<anonymous> (C:\Users\Farrukh Malik\AppData\Roaming\npm\node_modul
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)

Cannot find module 'internal/util/types' (CLI v1.7.14)

Your system information:

You have been opted out of telemetry. To change this, run: cordova telemetry on.
8.0.0

Ionic Version: 1.3.4
Ionic CLI Version: 1.7.14
Ionic App Lib Version: 0.7.0
OS:
Node Version: v8.9.3

now what should i do ??

**Comment by dshemin** [comment]
I found some solution and maybe it will help if you use 'n' for node version management.

1. Drop node to 6.2.2
2. Update npm
3. Switch to 8.* or 9.*
4. Should works

**Comment by dranreb06** [comment]
I fix the bug by modify **package.json.**
because i use on my local computer node version 9.3.0 and in the package.json it was 8.3.0 !
i simply change package.json with  
  "engines": {
    **"node": "9.3.0",**
    "npm": "5.5.1"
  },
in fact heroku use package.json to know wich version use to deploy.

**Comment by Butterfly** [comment]
@dranreb06 Congratulations on finding out a detail like that.

I wonder how many other users would be helped here on Github *right now* by being alerted to verify that detail.

Deleting that file and starting over has helped users a number of times so I also can't help wondering how many times that deleting that file first and foremost, if not only, deleted *that* discrepancy.

I *FURTHER* wonder how that discrepancy comes into existence in the first place, *and then*... how can it be easily detected from now on such that an error message automatically advises users to check that off their list as a causative.

As a how it might happen, maybe it's about how we can move around within the directory tree as we do various things over time. If mobility within the directory tree is one piece of the puzzle, I fall back on that it would be nice if `npm` checked that point specifically and then alerted specifically about it if that's the only problem..

**Comment by mcornetto** [comment]
If you go the delete and reinstall route, you also need to remove the npm and npm_cache directory in AppData/Roaming.

**Comment by gusarovecejlo** [comment]
Got this error when i call `gulp`, i was try to fix with node 8x, 9x, npm 5.8.0/5.6.0, but helped only reinstall local gulp, so `graceful-fs` one of the part of local `node_modules`, but i dont know for what in here, current v `graceful-fs@4.1.11`

**Comment by benmirkhah** [comment]
Switched back to node v6.9.5 to get around this issue!
